lib/chat/utils/tagging/tagging_textfield_ta.dart

```diff
@@ -0,0 +1,250 @@
+  final InputDecoration? decoration;
+  final TextStyle? style;
+  final Function(String)? onChange;
+  final int chatroomId;
+
+  const TaggingAheadTextField({
+    super.key,
+    required this.isDown,
+    required this.chatroomId,
+    required this.onTagSelected,
+    required this.controller,
+    required this.focusNode,
+    this.style,
+    this.decoration,
+    this.onChange,
+  });
+
+  @override
+  State<TaggingAheadTextField> createState() => _TaggingAheadTextFieldState();
+}
+
+class _TaggingAheadTextFieldState extends State<TaggingAheadTextField> {
+  late final TextEditingController _controller;
+  FocusNode? _focusNode;
+  LMBranding lmBranding = LMBranding.instance;
+  final ScrollController _scrollController = ScrollController();
+  final SuggestionsBoxController _suggestionsBoxController =
+      SuggestionsBoxController();
+
+  List<UserTag> userTags = [];
+
+  int page = 1;
+  int tagCount = 0;
+  bool tagComplete = false;
+  String textValue = "";
+  String tagValue = "";
+  static const FIXED_SIZE = 6;
+
+  @override
+  void initState() {
+    super.initState();
+    _focusNode = widget.focusNode;
+    _controller = widget.controller!;
+    _scrollController.addListener(() async {
+      // page++;
+      if (_scrollController.position.pixels ==
+          _scrollController.position.maxScrollExtent) {
+        page++;
+        final taggingData = (await locator<LikeMindsService>().getTaggingList(
+          (TagRequestModelBuilder()
+                ..chatroomId(widget.chatroomId)
+                ..page(page)
+                ..pageSize(FIXED_SIZE))
+              .build(),
+        ))
+            .data;
+        if (taggingData!.members != null && taggingData.members!.isNotEmpty) {
+          userTags.addAll(taggingData.members!.map((e) => e).toList());
+          // return userTags;
+        }
+      }
+    });
+  }
+
+  TextEditingController? get controller => _controller;
+
+  FutureOr<Iterable<UserTag>> _getSuggestions(String query) async {
+    String currentText = query;
+    try {
+      if (currentText.isEmpty) {
+        return const Iterable.empty();
+      } else if (!tagComplete && currentText.contains('@')) {
+        String tag = tagValue.substring(1).replaceAll(' ', '');
+        final taggingData = (await locator<LikeMindsService>().getTaggingList(
+          (TagRequestModelBuilder()
+                ..chatroomId(widget.chatroomId)
+                ..page(page)
+                ..searchQuery(tag)
+                ..pageSize(FIXED_SIZE))
+              .build(),
+        ))
+            .data;
+        if (taggingData!.members != null && taggingData.members!.isNotEmpty) {
+          userTags = taggingData.members!.map((e) => e).toList();
+          return userTags;
+        }
+        return const Iterable.empty();
+      } else {
+        return const Iterable.empty();
+      }
+    } catch (e) {
+      return const Iterable.empty();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
this `build()` is too big, divide this function into smaller fucntions and comments after each operations.